src/virtio/block.c

```diff
@@ -0,0 +1,463 @@
+
+                // Converting virtio sector number to sddf block number, we are rounding down
+                uint32_t sddf_block_number = (virtio_req->sector * VIRTIO_BLK_SECTOR_SIZE) / BLK_TRANSFER_SIZE;
+                // Converting bytes to the number of blocks, we are rounding up
+                uint16_t sddf_count = (virtq->desc[curr_desc_head].len + BLK_TRANSFER_SIZE - 1) / BLK_TRANSFER_SIZE;
+
+                // Check if request can be handled
+                if (!sddf_make_req_check(queue_handle, sddf_count)) {
+                    virtio_blk_set_req_fail(dev, desc_head);
+                    has_dropped = true;
+                    break;
+                }
+                
+                // Allocate data buffer from data region based on sddf_count
+                uintptr_t sddf_data;
+                fsmem_alloc(&fsmem_data, &sddf_data, sddf_count);
+                
+                // Bookkeep the virtio sddf block size translation
+                uintptr_t virtio_data = sddf_data + (virtio_req->sector * VIRTIO_BLK_SECTOR_SIZE) % BLK_TRANSFER_SIZE;
+                uintptr_t virtio_data_size = virtq->desc[curr_desc_head].len;
+
+                // Book keep the request
+                data_t data = {desc_head, sddf_data, sddf_count, sddf_block_number, virtio_data, virtio_data_size, false};
+                uint64_t req_id;
+                datastore_alloc(&datastore, &data, &req_id);
+                
+                // Pass this allocated data buffer to sddf read request, then enqueue it
+                blk_enqueue_req(queue_handle, READ_BLOCKS, sddf_data, sddf_block_number, sddf_count, req_id);
+                break;
+            }
+            case VIRTIO_BLK_T_OUT: {
+                LOG_BLOCK("Request type is VIRTIO_BLK_T_OUT\n");
+                LOG_BLOCK("Sector (read/write offset) is %d\n", virtio_req->sector);
+                
+                curr_desc_head = virtq->desc[curr_desc_head].next;
+                LOG_BLOCK("Descriptor index is %d, Descriptor flags are: 0x%x, length is 0x%x\n", curr_desc_head, (uint16_t)virtq->desc[curr_desc_head].flags, virtq->desc[curr_desc_head].len);
+
+                // Converting virtio sector number to sddf block number, we are rounding down
+                uint32_t sddf_block_number = (virtio_req->sector * VIRTIO_BLK_SECTOR_SIZE) / BLK_TRANSFER_SIZE;
+                // Converting bytes to the number of blocks, we are rounding up
+                uint16_t sddf_count = (virtq->desc[curr_desc_head].len + BLK_TRANSFER_SIZE - 1) / BLK_TRANSFER_SIZE;
+                
+                bool not_aligned = ((virtio_req->sector % (BLK_TRANSFER_SIZE / VIRTIO_BLK_SECTOR_SIZE)) != 0);
+
+                // @ericc: If the write request is not aligned to the sddf block size, we need to first read the surrounding aligned memory, overwrite that 
```

Comment:
Should this be prefixed with `@ericc` or is just a general comment?

Reply:
Just a general comment, will change it up